pkgs/development/tools/misc/edb/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, boost, capstone
+, double-conversion, graphviz, qtxmlpatterns }:
+
+stdenv.mkDerivation rec {
+  pname = "edb";
+  version = "2019-04-04";
+
+  src = fetchFromGitHub {
+    owner = "eteran";
+    repo = "edb-debugger";
+    rev = "77e5ebf497ff489827fe1b4b5861a77688aa3729";
+    fetchSubmodules = true;
+    sha256 = "1pfz65q5p1cpr2jz8mwpdhyckk7rx21inha4a65ykpim745n18c5";
```

Comment:
When building I got:
```
hash mismatch in fixed-output derivation '/nix/store/36lqk5j7zcmi05dc8rfyzhiznfzfh0q9-source':
  wanted: sha256:1pfz65q5p1cpr2jz8mwpdhyckk7rx21inha4a65ykpim745n18c5
  got:    sha256:1mmw0pgb1zgqsjzgshrpw988higzbm84wk8k6kaqahxwh131xq1y
```
The `fetchSubmodule` option is problematic because it is hard to make it deterministic. It would be nice if you find a way to build qhexview as a separate derivation, and use it as a normal dependency instead of a submodule. Then the non-determinism would disappear. Of course, depending on the build system of edb, this may or may not be feasible.

Reply:
Ok. I will have a go at packaging qhexview as a separate derivation.

Reply:
I don't think it is worth packaging [qhexview](https://github.com/eteran/qhexview) seperately as it only contains a single `.cpp` and `.h` file.

AFAICT the non-determinism is caused by `leaveDotGit` not `fetchSubmodules` (see #8567).
Therefore, I have removed the `leaveDotGit` option and instead create the `.git` directory in `postPatch`. This is the same approach taken in the hotspot package: https://github.com/NixOS/nixpkgs/blob/43cfb6e570207b5960f8ee115e85689880b82bb0/pkgs/development/tools/analysis/hotspot/default.nix#L42-L48

Reply:
Ah sorry for having misled you about  `fetchSubmodules`. I stand corrected. It is nice that you found a way out.